Fix: coverage PR comment fails for fork / external contributor PRs

## Problem

ref https://github.com/prest/prest/actions/runs/29113420319/job/86431112991?pr=975
ref https://github.com/prest/prest/pull/975

The **Coverage PR comment** job in [`.github/workflows/test-unit.yml`](.github/workflows/test-unit.yml) fails on pull requests from forks (external contributors):

```
GraphQL: Resource not accessible by integration (addComment)
Error: Process completed with exit code 1.
```

Example: [PR #975](https://github.com/prest/prest/pull/975) — job `coverage-report` (uses `fgrosse/go-coverage-report@v1.3.0`).

Unit tests pass; only the PR comment step fails. This blocks or marks CI red for contributors who cannot write to the base repo.

## Cause

GitHub restricts the default `GITHUB_TOKEN` on **`pull_request` events from forks**. Even with `pull-requests: write` in the workflow, the token cannot post PR comments for security reasons.

## Proposed fix

Pick one (or combine):

1. **Skip comment on fork PRs** (simplest):
   ```yaml
   coverage-report:
     if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
   ```

2. **Do not fail the workflow when comment is denied** — e.g. `continue-on-error: true` on the coverage-report job, or catch failure in the action wrapper.

3. **Fallback for forks** — write coverage delta to the **job summary** (`$GITHUB_STEP_SUMMARY`) instead of a PR comment, so contributors still see results in the Actions UI.

## Acceptance criteria

- [ ] Fork PRs: unit tests run; CI is not failed solely because the coverage comment could not be posted
- [ ] Same-repo PRs: coverage comment still posted when possible
- [ ] Document in CONTRIBUTING.md that fork PRs may not get an automated coverage comment (if skipped)

## Non-goals

- Switching to `pull_request_target` (runs untrusted code with base-repo credentials — avoid unless carefully scoped)
- Requiring external contributors to use branches in the main repo

## References

- Workflow: `.github/workflows/test-unit.yml` → job `coverage-report`
- GitHub docs: [Workflow permissions for fork PRs](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)

--- a/.github/workflows/test-unit.yml
+++ b/.github/workflows/test-unit.yml
@@ -42,7 +42,7 @@ jobs:
 
   coverage-report:
     name: Coverage PR comment
-    if: github.event_name == 'pull_request'
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name == github.repository
     runs-on: ubuntu-latest
     needs: unit
     permissions:
@@ -57,3 +57,43 @@ jobs:
           root-package: github.com/prest/prest/v2
           trim: github.com/prest/prest/v2/
           github-baseline-workflow-ref: test-unit.yml
+
+  coverage-report-fork:
+    name: Coverage PR comment (fork)
+    if: github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository
+    runs-on: ubuntu-latest
+    needs: unit
+    permissions:
+      contents: read
+      actions: read
+    steps:
+      - uses: fgrosse/go-coverage-report@v1.3.0
+        id: coverage
+        with:
+          skip-comment: true
+          coverage-artifact-name: code-coverage
+          coverage-file-name: coverage.out
+          root-package: github.com/prest/prest/v2
+          trim: github.com/prest/prest/v2/
+          github-baseline-workflow-ref: test-unit.yml
+
+      # Fork PRs: GITHUB_TOKEN cannot comment on the base repo; use PR_TOKEN (PAT).
+      - name: Post coverage comment
+        env:
+          GH_TOKEN: ${{ secrets.PR_TOKEN }}
+          COVERAGE_REPORT: ${{ steps.coverage.outputs.coverage_report }}
+        run: |
+          if [ -z "$COVERAGE_REPORT" ]; then
+            echo "No coverage report to post"
+            exit 0
+          fi
+
+          printf '%s\n' "$COVERAGE_REPORT" > coverage-comment.md
+
+          COMMENT_ID=$(gh api "repos/${{ github.repository }}/issues/${{ github.event.pull_request.number }}/comments" \
+            -q '.[] | select(.body | test("Coverage Δ")) | .id' | head -n 1)
+          if [ -n "$COMMENT_ID" ]; then
+            gh api -X DELETE "repos/${{ github.repository }}/issues/comments/${COMMENT_ID}" || true
+          fi
+
+          gh pr comment "${{ github.event.pull_request.number }}" --body-file=coverage-comment.md

